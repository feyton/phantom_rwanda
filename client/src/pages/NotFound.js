import React from 'react';
import { Link } from 'react-router-dom';

function NotFound() {
  return (
    <div className="flex items-center flex-col justify-center h-screen">
      <h1 className="text-bold text-3xl font-rale">Oops. Page Not Found</h1>
      <Link
        to="/"
        className="py-1 px-2 bg-primary hover:bg-hover font-bold rounded-md my-3 text-white"
      >
        Back Home
      </Link>
    </div>
  );
}

export default NotFound;
