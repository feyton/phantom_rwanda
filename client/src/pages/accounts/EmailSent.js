import { PrimaryButton } from '@components/Button.js';
import React from 'react';
import { Link, useLocation, useParams } from 'react-router-dom';

const img =
  'https://cdn.pixabay.com/photo/2012/04/18/19/01/check-37583_960_720.png';

function EmailSent() {
  const location = useLocation();
  return (
    <>
      <h3 className="text-2xl font-rale mb-2 text-center font-bold">
        Email Sent
      </h3>
      <img src={img} alt="" className="h-24  mx-auto mt-4" />

      <div className="text-left mt-4">
        <p className="mb-2 text-justify text-sm">
          A link to reset your password has been sent to{' '}
          <b>{location?.state?.email}</b>
        </p>

        <p className=" text-justify text-sm">
          Did not receive the email? Remember to check <b>Junk/Spam</b> folder
        </p>
      </div>

      <Link to="/" className="w-fit my-2">
        <PrimaryButton name="Home" />
      </Link>
    </>
  );
}

export default EmailSent;
