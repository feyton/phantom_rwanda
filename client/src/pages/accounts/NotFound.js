import React from 'react';
import { Link } from 'react-router-dom';
import { Button } from './AccountRouter.js';

function NotFoundAccount() {
  return (
    <div className="w-fit flex flex-col justify-center items-center p-3">
      <h3 className="text-3xl font-bold pb-4">Oops... Not Found</h3>
      <Link to="/">
        <Button name="Home" styles=" bg-primary mt-2" />
      </Link>
    </div>
  );
}

export default NotFoundAccount;
