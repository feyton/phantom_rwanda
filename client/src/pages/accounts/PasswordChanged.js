import React from 'react';
import { Link } from 'react-router-dom';

function PasswordChanged() {
  return (
    <>
      <h3 className="text-2xl text-center font-rale font-bold mb-2">
        Password Changed
      </h3>
      <hr />
      <div className="text-left">
        <p className="py-1 mb-2  text-sm">
          Dear, you got yourself a new password.
        </p>
        <Link
          to="/login"
          className="py-1 px-3  bg-primary hover:bg-hover text-white rounded-md font-bold mt-5 mx-auto"
        >
          Login
        </Link>

        <p className="py-1 mt-2 text-sm">
          You will receive email confirming your changes.
        </p>
      </div>
    </>
  );
}

export default PasswordChanged;
