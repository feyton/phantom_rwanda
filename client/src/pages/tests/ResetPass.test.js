import { fireEvent, render, screen } from '@testing-library/react';
import mockAxios from 'axios';
import { mount, shallow } from 'enzyme';
import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer, { act } from 'react-test-renderer';
import ResetRouterLayout from '../accounts/AccountRouter.js';
import ResetFormPage from '../accounts/EmailForm.js';
import EmailSent from '../accounts/EmailSent.js';
import ResetNewPassword from '../accounts/NewPasswordForm.js';
import PasswordChanged from '../accounts/PasswordChanged.js';
import NotFound from '../NotFound.js';

const mockHandleEmail = jest.fn((email) => {
  return Promise.resolve(email);
});

const wrapper = shallow(
  <BrowserRouter>
    <ResetRouterLayout />
  </BrowserRouter>
);
describe('function testing', () => {
  let wrapper;

  it('Should return a errors', async () => {
    await act(async () => {
      wrapper = mount(
        <BrowserRouter>
          <ResetFormPage handleEmail={mockHandleEmail} />
        </BrowserRouter>
      );
      wrapper.find('form').simulate('submit');
      expect(await wrapper).toMatchSnapshot();
    });
  });
});

describe('Reset page layout', () => {
  it('Render without crashing', () => {
    expect(wrapper.length).toEqual(1);
  });
});

describe('Reset password layout', () => {
  it('should render all routes', () => {
    const elem = renderer
      .create(
        <BrowserRouter>
          <ResetRouterLayout />
        </BrowserRouter>
      )
      .toJSON();
    expect(elem).toMatchSnapshot();
  });
  it('should render the new password form', () => {
    const elem = renderer
      .create(
        <BrowserRouter>
          <ResetNewPassword />
        </BrowserRouter>
      )
      .toJSON();
    expect(elem).toMatchSnapshot();
  });
  it('should render the email sent notification', () => {
    const elem = renderer
      .create(
        <BrowserRouter>
          <EmailSent />
        </BrowserRouter>
      )
      .toJSON();
    expect(elem).toMatchSnapshot();
  });
  it('should render the email Form', () => {
    const elem = renderer
      .create(
        <BrowserRouter>
          <ResetNewPassword />
        </BrowserRouter>
      )
      .toJSON();
    expect(elem).toMatchSnapshot();
  });

  it('should render the success message', () => {
    const elem = renderer
      .create(
        <BrowserRouter>
          <PasswordChanged />
        </BrowserRouter>
      )
      .toJSON();
    expect(elem).toMatchSnapshot();
  });
});

describe('NotFound', () => {
  it('It renders without', () => {
    const elem = renderer.create(
      <BrowserRouter>
        <NotFound />
      </BrowserRouter>
    );
    expect(elem).toMatchSnapshot();
  });
});
describe('Testing Library', () => {
  let elem;
  beforeEach(() => {
    elem = render(
      <BrowserRouter>
        <ResetFormPage handleEmail={mockHandleEmail} />
      </BrowserRouter>
    );
  });
  it('Should display required error when email is invalid', async () => {
    fireEvent.submit(screen.getByTestId('form'));
    expect(await screen.getAllByRole('alert')).toHaveLength(1);
    expect(mockHandleEmail).not.toBeCalled();
  });
  it('Should display an invalid email error', async () => {
    fireEvent.input(screen.getByRole('textbox', { name: /email/i }), {
      target: {
        value: 'test'
      }
    });
    fireEvent.submit(screen.getByTestId('form'));
    expect(await screen.findAllByRole('alert')).toHaveLength(1);
    expect(await screen.findAllByText('Invalid email')).toHaveLength(1);
  });
  it('Should submit the form when email is valid', async () => {
    fireEvent.input(screen.getByRole('textbox', { name: /email/i }), {
      target: {
        value: 'fab@me.com'
      }
    });
    fireEvent.submit(screen.getByTestId('form'));
    mockAxios.post.mockImplementationOnce(() => {
      return Promise.reject({
        status: 400,
        data: { data: { message: 'Email not found' } }
      });
    });
    expect(await screen.findAllByText('Sending')).toHaveLength(1);
  });
  it('Should submit the form and raise an error when email not found', async () => {
    fireEvent.input(screen.getByRole('textbox', { name: /email/i }), {
      target: {
        value: 'fab@me.com'
      }
    });
    fireEvent.submit(screen.getByTestId('form'));
    expect(await screen.findAllByText('Sending')).toHaveLength(1);
    mockAxios.post.mockImplementationOnce(() => {
      return Promise.resolve({
        status: 200,
        data: { data: { message: 'Link is sent' } }
      });
    });
    expect(elem).toMatchSnapshot();
  });
});
