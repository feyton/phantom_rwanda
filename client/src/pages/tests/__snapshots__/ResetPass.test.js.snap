// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound It renders without 1`] = `
<div
  className="flex items-center flex-col justify-center h-screen"
>
  <h1
    className="text-bold text-3xl font-rale"
  >
    Oops. Page Not Found
  </h1>
  <a
    className="py-1 px-2 bg-primary hover:bg-hover font-bold rounded-md my-3 text-white"
    href="/"
    onClick={[Function]}
  >
    Back Home
  </a>
</div>
`;

exports[`Reset password layout should render all routes 1`] = `
<div
  className="min-h-[75vh] h-full flex flex-col pb-40 md:pb-2 relative py-4 md:flex-row items-center justify-center"
>
  <div
    className="max-w-sm mr-10 mt-[-60px] md:flex md:items-center md:justify-center md:mx-auto md:my-[auto] xl:ml-64"
  >
    <img
      alt="placeholder-img"
      className="w-full object-cover"
      src="https://res.cloudinary.com/feyton/image/upload/v1645616111/Codebandits/Phantom_icon_1_tpjkws.png"
    />
  </div>
  <div
    className="flex flex-col  py-3 px-4 w-72 md:w-80 md:min-w-[300px] md:max-w-sm shadow-main absolute bg-white top-16 md:top-0 md:mx-auto md:relative rounded-lg xl:mr-64"
  >
    <div
      className="w-fit flex flex-col justify-center items-center p-3"
    >
      <h3
        className="text-3xl font-bold pb-4"
      >
        Oops... Not Found
      </h3>
      <a
        href="/"
        onClick={[Function]}
      >
        <div>
          <button
            className="rounded-md transition-all text-white py-1 px-5 mt-2 bg-primary hover:bg-hover hover:transition-all font-bold  bg-primary mt-2"
            type="button"
          >
            Home
          </button>
        </div>
      </a>
    </div>
  </div>
</div>
`;

exports[`Reset password layout should render the email Form 1`] = `
<div>
   
  <h3
    className="text-2xl font-bold text-center py-3 font-rale"
  >
    You are not authorized
  </h3>
  <p
    className="mb-3 text-sm"
  >
    You don't have access or your access has expired
  </p>
  <div />
  <a
    href="/accounts/reset-password"
    onClick={[Function]}
  >
    Try again
  </a>
  <a
    href="/"
    onClick={[Function]}
  >
    Cancel
  </a>
</div>
`;

exports[`Reset password layout should render the email sent notification 1`] = `
Array [
  <h3
    className="text-2xl font-rale mb-2 text-center font-bold"
  >
    Email Sent
  </h3>,
  <img
    alt=""
    className="h-24  mx-auto mt-4"
    src="https://cdn.pixabay.com/photo/2012/04/18/19/01/check-37583_960_720.png"
  />,
  <div
    className="text-left mt-4"
  >
    <p
      className="mb-2 text-justify text-sm"
    >
      A link to reset your password has been sent to
       
      <b />
    </p>
    <p
      className=" text-justify text-sm"
    >
      Did not receive the email? Remember to check 
      <b>
        Junk/Spam
      </b>
       folder
    </p>
  </div>,
  <a
    className="w-fit my-2"
    href="/"
    onClick={[Function]}
  >
    <button
      className="bg-primary rounded-md text-white font-bold font-raleway py-2 px-3 hover:bg-hover transition-all block hover:transition-all"
      type="button"
    >
      Home
    </button>
  </a>,
]
`;

exports[`Reset password layout should render the new password form 1`] = `
<div>
   
  <h3
    className="text-2xl font-bold text-center py-3 font-rale"
  >
    You are not authorized
  </h3>
  <p
    className="mb-3 text-sm"
  >
    You don't have access or your access has expired
  </p>
  <div />
  <a
    href="/accounts/reset-password"
    onClick={[Function]}
  >
    Try again
  </a>
  <a
    href="/"
    onClick={[Function]}
  >
    Cancel
  </a>
</div>
`;

exports[`Reset password layout should render the success message 1`] = `
Array [
  <h3
    className="text-2xl text-center font-rale font-bold mb-2"
  >
    Password Changed
  </h3>,
  <hr />,
  <div
    className="text-left"
  >
    <p
      className="py-1 mb-2  text-sm"
    >
      Dear, you got yourself a new password.
    </p>
    <a
      className="py-1 px-3  bg-primary hover:bg-hover text-white rounded-md font-bold mt-5 mx-auto"
      href="/login"
      onClick={[Function]}
    >
      Login
    </a>
    <p
      className="py-1 mt-2 text-sm"
    >
      You will receive email confirming your changes.
    </p>
  </div>,
]
`;

exports[`Testing Library Should submit the form and raise an error when email not found 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h3
        class="text-2xl font-bold text-center py-3 font-rale"
      >
        RESET PASSWORD
      </h3>
      <p
        class="mb-3 text-sm"
      >
        Enter your email to receive a password reset link
      </p>
      <form
        id="form"
        role="form"
      >
        <div
          class="relative w-full"
        >
          <label
            class="block font-bold mb-2 focus:outline-none required"
            for="email"
          >
            Email:
          </label>
          <input
            class="block w-full rounded-md border-2 placeholder:text-gray-400 placeholder:text-small py-1 px-3 focus:outline-none border-green-500 "
            id="email"
            name="email"
            placeholder="Your email"
            type="email"
          />
          <p
            class="text-sm text-red-900 font-bold py-1/2 pl-2 opacity-0"
            role="alert"
          />
        </div>
        <div
          class="flex flex-row justify-between mt-2"
        >
          <div>
            <button
              class="rounded-md transition-all text-white py-1 px-5 mt-2 bg-primary hover:bg-hover hover:transition-all font-bold undefined"
              type="submit"
            >
              SUBMIT
            </button>
          </div>
          <a
            href="/"
          >
            <div>
              <button
                class="rounded-md transition-all text-white py-1 px-5 mt-2 bg-red-500 hover:bg-hover hover:transition-all font-bold hover:bg-hovercancel bg-cancel hover:transition-all"
                type="button"
              >
                CANCEL
              </button>
            </div>
          </a>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <h3
      class="text-2xl font-bold text-center py-3 font-rale"
    >
      RESET PASSWORD
    </h3>
    <p
      class="mb-3 text-sm"
    >
      Enter your email to receive a password reset link
    </p>
    <form
      id="form"
      role="form"
    >
      <div
        class="relative w-full"
      >
        <label
          class="block font-bold mb-2 focus:outline-none required"
          for="email"
        >
          Email:
        </label>
        <input
          class="block w-full rounded-md border-2 placeholder:text-gray-400 placeholder:text-small py-1 px-3 focus:outline-none border-green-500 "
          id="email"
          name="email"
          placeholder="Your email"
          type="email"
        />
        <p
          class="text-sm text-red-900 font-bold py-1/2 pl-2 opacity-0"
          role="alert"
        />
      </div>
      <div
        class="flex flex-row justify-between mt-2"
      >
        <div>
          <button
            class="rounded-md transition-all text-white py-1 px-5 mt-2 bg-primary hover:bg-hover hover:transition-all font-bold undefined"
            type="submit"
          >
            SUBMIT
          </button>
        </div>
        <a
          href="/"
        >
          <div>
            <button
              class="rounded-md transition-all text-white py-1 px-5 mt-2 bg-red-500 hover:bg-hover hover:transition-all font-bold hover:bg-hovercancel bg-cancel hover:transition-all"
              type="button"
            >
              CANCEL
            </button>
          </div>
        </a>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`function testing Should return a errors 1`] = `
<BrowserRouter>
  <Router
    location={
      Object {
        "hash": "",
        "key": "default",
        "pathname": "/",
        "search": "",
        "state": null,
      }
    }
    navigationType="POP"
    navigator={
      Object {
        "action": "POP",
        "back": [Function],
        "block": [Function],
        "createHref": [Function],
        "forward": [Function],
        "go": [Function],
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "default",
          "pathname": "/",
          "search": "",
          "state": null,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <ResetFormPage
      handleEmail={[MockFunction]}
    >
      <h3
        className="text-2xl font-bold text-center py-3 font-rale"
      >
        RESET PASSWORD
      </h3>
      <p
        className="mb-3 text-sm"
      >
        Enter your email to receive a password reset link
      </p>
      <form
        id="form"
        onSubmit={[Function]}
        role="form"
      >
        <div
          className="relative w-full"
        >
          <label
            className="block font-bold mb-2 focus:outline-none required"
            htmlFor="email"
          >
            Email:
          </label>
          <input
            className="block w-full rounded-md border-2 placeholder:text-gray-400 placeholder:text-small py-1 px-3 focus:outline-none border-green-500 "
            id="email"
            name="email"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Your email"
            type="email"
          />
          <p
            className="text-sm text-red-900 font-bold py-1/2 pl-2 opacity-0"
            role="alert"
          />
        </div>
        <div
          className="flex flex-row justify-between mt-2"
        >
          <Button
            name="SUBMIT"
            type="submit"
          >
            <div>
              <button
                className="rounded-md transition-all text-white py-1 px-5 mt-2 bg-primary hover:bg-hover hover:transition-all font-bold undefined"
                type="submit"
              >
                SUBMIT
              </button>
            </div>
          </Button>
          <Link
            to="/"
          >
            <a
              href="/"
              onClick={[Function]}
            >
              <Button
                icon="cancel"
                name="CANCEL"
                styles="hover:bg-hovercancel bg-cancel hover:transition-all"
                type="button"
              >
                <div>
                  <button
                    className="rounded-md transition-all text-white py-1 px-5 mt-2 bg-red-500 hover:bg-hover hover:transition-all font-bold hover:bg-hovercancel bg-cancel hover:transition-all"
                    type="button"
                  >
                    CANCEL
                  </button>
                </div>
              </Button>
            </a>
          </Link>
        </div>
      </form>
    </ResetFormPage>
  </Router>
</BrowserRouter>
`;
